File: Archive - Java/PoseDetection/app/src/main/java/com/icanerdogan/posedetection/MainActivity2.java
package com.icanerdogan.posedetection;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity2 extends AppCompatActivity implements View.OnClickListener {
    ImageView imageViewPose;
    LinearLayout btnRefresh;
    TextView angleTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        imageViewPose = findViewById(R.id.imageView2);
        angleTextView = findViewById(R.id.angleText);
        btnRefresh = findViewById(R.id.refresh);

        btnRefresh.setOnClickListener(this);

        Intent intent = getIntent();
        String text = intent.getStringExtra("Text");
        angleTextView.setText(text);

        Singleton singleton = Singleton.getInstance();
        imageViewPose.setImageBitmap(singleton.getMyImage());
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.refresh:
                backMain();
                break;
        }
    }

    private void backMain() {
        Intent intent = new Intent(MainActivity2.this, MainActivity.class);
        startActivity(intent);
    }
}

File: Kotlin/TextRecognitionApp/app/src/main/java/com/ibrahimcanerdogan/textrecognitionapp/MainActivity2.kt
package com.ibrahimcanerdogan.textrecognitionapp

import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.os.Build
import android.os.Build.VERSION
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.ibrahimcanerdogan.textrecognitionapp.databinding.ActivityMain2Binding

class MainActivity2 : AppCompatActivity() {

    private lateinit var binding : ActivityMain2Binding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMain2Binding.inflate(layoutInflater)
        setContentView(binding.root)

        val (bitmap, uri) = getImageFormatFromIntent()
        setImage(uri, bitmap)

        binding.buttonBack.setOnClickListener {
            onBackPressed()
        }
    }

    private fun setImage(uri: String?, bitmap: Bitmap?) {
        binding.imageView.setImageDrawable(null)
        if (uri != null) {
            binding.imageView.setImageURI(Uri.parse(uri))
        } else {
            binding.imageView.setImageBitmap(bitmap)
        }
    }

    private fun getImageFormatFromIntent() : Pair<Bitmap?, String?> {
        val intent : Intent = getIntent()

        val bitmap = if (VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) intent.getParcelableExtra("bitmapimage", Bitmap::class.java)
            else intent.getParcelableExtra("bitmapimage")

        val uri = intent.getStringExtra("uriimage")

        return Pair(bitmap, uri)
    }
}